### PR TITLE
follow secure coding BP, fix (#993)

### PR DESCRIPTION
**Pull request by bayotop**
I created a fresh PR for #993. I am quite certain the changes shouldn't affect other tests so I am not sure why they are still failing because of a missing API key. 

The tests related to constructing the event are passing. Please let me know if there is anything I can do. Thanks.

### CODE REVIEW

**Comment by ob-stripe** [comment]
Ah, I believe I understand what's happening. The `STRIPE_TEST_SK` environment variable is defined as a "secure variable" in our [`appveyor.yml`](https://github.com/stripe/stripe-dotnet/blob/a43bc26f5d91d94e049bfc9c307b017f4b1c3ded/appveyor.yml#L5-L6) configuration file.

From https://www.appveyor.com/docs/build-configuration/#secure-variables:
> However, secure variables are not decoded during Pull Request builds which prevents someone from submitting PR with malicious build script displaying those variables. In more controlled environment through with a trusted team and private GitHub repositories there is an option on General tab of project settings to allow secure variables for PRs.

We should probably just define the key as a regular variable -- the "secret" API key for the test account we use for our libraries is visible in plain text in many of our other language libraries (e.g. [here](https://github.com/stripe/stripe-python/blob/ef50aa5e96b05a47f4b277ac6eab76e336860b99/stripe/test/helper.py#L132-L133) in stripe-python).

File: src/Stripe.Tests.XUnit/Stripe.Tests.XUnit.csproj [hunk 1]
```diff
@@ -27,4 +27,8 @@
     <ProjectReference Include="..\Stripe.net\Stripe.net.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
```
Comment (brandur-stripe) [inline src/Stripe.Tests.XUnit/Stripe.Tests.XUnit.csproj]: Do you know why these ended up in here? Can we pull them out of the branch for hygiene?

File: src/Stripe.net.Tests/Stripe.net.Tests.csproj [hunk 2]
```diff
@@ -49,4 +49,8 @@
     <Reference Include="System" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
```
Comment (brandur-stripe) [inline src/Stripe.net.Tests/Stripe.net.Tests.csproj]: Same with this one.

**Comment by brandur-stripe** [comment]
Nice one @bayotop! Thanks again. I left a couple minor comments, but this looks great to me.

@ob-stripe Would you mind taking a review pass as well? You know this code a lot better than I do.

> Ah, I believe I understand what's happening. The STRIPE_TEST_SK environment variable is defined as a "secure variable" in our appveyor.yml configuration file.
>
> We should probably just define the key as a regular variable -- the "secret" API key for the test account we use for our libraries is visible in plain text in many of our other language libraries (e.g. here in stripe-python).

Ah yep, Travis does this too :/ I don't think there's a major problem putting a secret testmode key into here as plaintext — it does allow someone to potentially DOS our builds, but the advantages of having successful builds from third-party contributors outweights that by a fair bit.

In the meantime, we can just repush @bayotop's branch to our own repository just to get this unblocked.

File: src/Stripe.net.Tests/Stripe.net.Tests.csproj (hunk 2 above)
Comment (bayotop) [reply src/Stripe.net.Tests/Stripe.net.Tests.csproj]: Good point. Those were added during the build. However, this isn't the case anymore and it seems they can be safely removed. I'll amend the commit.

**Comment by bayotop** [comment]
I'll amended the commit according to you comments and aligned with master. Feel free to proceed with the repush. Thanks!

**Comment by bayotop** [comment]
My event.json file had Windows line endings so produced a different hash then expected, thus the 2 failed tests on appveyor/branch. Should be fixed now.

**Comment by brandur-stripe** [comment]
> My local event.json file had Windows line endings so produced a different hash then expected, thus the 2 failed tests on appveyor/branch. Should be fixed now.

Thanks for taking a look at that Martin! I've repushed to our origin again and will get this in as soon as the build passes.

File: src/Stripe.net.Tests/Stripe.net.Tests.csproj (hunk 2 above)
Comment (xt0rted) [reply src/Stripe.net.Tests/Stripe.net.Tests.csproj]: These are added by VS to indicate the given project is a test project and that the test runner should be loaded for it.

**Pull request merged**